From: Filippo Giunchedi <filippo@wikimedia.org>
Date: Thu, 16 Nov 2017 15:16:08 +0100
Subject: Disable cloudfoundry support

Change-Id: Id4f74d3a02012e9b5f168037838d9f4ffac33096
---
 main.go | 35 -----------------------------------
 1 file changed, 35 deletions(-)

diff --git a/main.go b/main.go
index 460e802..26fbf7c 100644
--- a/main.go
+++ b/main.go
@@ -9,7 +9,6 @@ import (
 	"strings"
 
 	log "github.com/Sirupsen/logrus"
-	"github.com/cloudfoundry-community/go-cfenv"
 	"github.com/oliver006/redis_exporter/exporter"
 	"github.com/prometheus/client_golang/prometheus"
 )
@@ -27,7 +26,6 @@ var (
 	isDebug       = flag.Bool("debug", false, "Output verbose debug information")
 	logFormat     = flag.String("log-format", "txt", "Log format, valid options are txt and json")
 	showVersion   = flag.Bool("version", false, "Show version information and exit")
-	useCfBindings = flag.Bool("use-cf-bindings", false, "Use Cloud Foundry service bindings")
 	addrs         []string
 	passwords     []string
 	aliases       []string
@@ -73,8 +71,6 @@ func main() {
 		if err != nil {
 			log.Fatal(err)
 		}
-	case *useCfBindings:
-		addrs, passwords, aliases = getCloudFoundryRedisBindings()
 	default:
 		addrs, passwords, aliases = loadRedisArgs(*redisAddr, *redisPassword, *redisAlias, *separator)
 	}
@@ -180,34 +176,3 @@ func getEnv(key string, defaultVal string) string {
 	}
 	return defaultVal
 }
-
-func getCloudFoundryRedisBindings() (addrs, passwords, aliases []string) {
-	if !cfenv.IsRunningOnCF() {
-		return
-	}
-
-	appEnv, err := cfenv.Current()
-	if err != nil {
-		log.Warnln("Unable to get current CF environment", err)
-		return
-	}
-
-	redisServices, err := appEnv.Services.WithTag("redis")
-	if err != nil {
-		log.Warnln("Error while getting redis services", err)
-		return
-	}
-
-	for _, redisService := range redisServices {
-		credentials := redisService.Credentials
-		addr := credentials["hostname"].(string) + ":" + credentials["port"].(string)
-		password := credentials["password"].(string)
-		alias := redisService.Name
-
-		addrs = append(addrs, addr)
-		passwords = append(passwords, password)
-		aliases = append(aliases, alias)
-	}
-
-	return
-}
